### PR TITLE
auro-avatar: new generator build

### PR DESCRIPTION
**Pull request by blackfalcon**
# Alaska Airlines Pull Request

**Fixes:** #7, #6 

## Summary:

The scope of this update is to address new features as well as migration to a new generator build.

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: .github/settings.yml [hunk 1]
```diff
@@ -1,8 +1,9 @@
 repository:
   # See https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#edit for all available settings.
+  # See https://github.com/apps/settings for app settings
 
   # A short description of the repository that will show up on GitHub
-  description: Custom web component for the purpose of displaying an avatar image.
+  description: Custom element that ....
```
Comment (jason-capsule42) [inline .github/settings.yml:6]: Add the description back in

File: package.json [hunk 2]
```diff
@@ -7,83 +7,85 @@
     "  3. Go to http://localhost:8000",
     "================================================================================"
   ],
-  "name": "@alaskaairux/auro-avatar",
-  "version": "2.1.0",
-  "description": "Auro custom auro-avatar element",
-  "author": "Product design and research",
+  "name": "@aurodesignsystem/auro-avatar",
+  "version": "0.0.0",
```
Comment (jason-capsule42) [inline package.json:11]: Version will get written during publish, right?

File: scripts/prepForBuild.js [hunk 3]
```diff
@@ -9,7 +9,7 @@ let copyFiles = async function() {
     console.log(`${bundle} was copied to ./build dir`);
   });
 
-  fs.copyFile(`./demo/css/style.css`, `./build/css/style.css`, (err) => {
+  fs.copyFile(`./demo/style.css`, `./build/style.css`, (err) => {
```
Comment (jason-capsule42) [inline scripts/prepForBuild.js:12]: Shouldn't this account for:

```suggestion
  fs.copyFile(`./demo/sass/style.scss`, `./build/style.scss`, (err) => {
```

File: src/style.scss [hunk 4]
```diff
@@ -78,20 +66,35 @@ img {
...
+}
+
+:host([type='md'][tail]),
+:host([md][tail]) {
+  width: var(--lg);
+
+  div {
+    width: var(--md);
   }
 }
```
Comment (jason-capsule42) [inline src/style.scss:99]: :host[lg] versions of this?

File: src/style.scss [hunk 5]
```diff
@@ -37,38 +42,21 @@
...
   width: var(--sm);
   height: var(--sm);
 }
 
-:host([md]) {
+:host([md]),
+:host([type='md']) {
   width: var(--md);
   height: var(--md);
 }
```
Comment (jason-capsule42) [inline src/style.scss:55]: :host[lg] versions of this?

File: demo/demo.md [hunk 6]
```diff
@@ -21,38 +21,22 @@ The default component presents itself as a circle and default image placement.
...
-  <auro-avatar code="anc"></auro-avatar>
-  <auro-avatar code="mke" sm></auro-avatar>
-</div>
-
-<auro-accordion lowProfile justifyRight>
-  <span slot="trigger">See code</span>
-
-  ```html
-  <auro-avatar code="anc"></auro-avatar>
-  <auro-avatar code="mke" sm></auro-avatar>
```
Comment (jason-capsule42) [inline demo/demo.md:55]: `code` is still in the API documentation but examples removed here?

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (blackfalcon) [reply package.json:11]: The version number comes from the tag version. This gets committed back from semantic release.

File: scripts/prepForBuild.js (hunk 3 above)
Comment (blackfalcon) [reply scripts/prepForBuild.js:12]: The Sass directory has been removed from the demo directory.

File: src/style.scss (hunk 4 above)
Comment (blackfalcon) [reply src/style.scss:99]: LG is the baseline. Sm and Md are modifiers.

File: src/style.scss (hunk 5 above)
Comment (blackfalcon) [reply src/style.scss:55]: See the previous comment.

**Comment by blackfalcon** [comment]
@jason-capsule42 updated completed and demo located https://avatar-migration-10.surge.sh/

File: demo/demo.md (hunk 6 above)
Comment (blackfalcon) [reply demo/demo.md:55]: this was addressed with the last update.

File: demo/demo.md [hunk 7]
```diff
@@ -104,13 +91,13 @@ For the airline title, simply add content to the `display` slot. See the example
   <auro-avatar tail="AS">
     <span slot="display">Alaska Airlines</span>
   </auro-avatar>
-  <auro-avatar tail="AA">
+  <auro-avatar tail="AA" type="lg">
```
[the remark this answers is not in the document]
Comment (blackfalcon) [reply demo/demo.md]: addressed with last update

File: demo/demo.md [hunk 8]
```diff
@@ -62,37 +46,40 @@ The auro-avatar supports auto generation of pre-defined images using the `code`
...
+<small>Note: The previous `md` boolean attributes are deprecated.</small>
 
 For the airline title, simply add content to the `display` slot. See the example code for the HTML API.
 
 <div class="exampleWrapper">
   <auro-avatar tail="AS">
     <span slot="display">Alaska Airlines</span>
   </auro-avatar>
-  <auro-avatar tail="AA">
+  <auro-avatar tail="AA" type="lg">
```
[the remark this answers is not in the document]
Comment (blackfalcon) [reply demo/demo.md]: addressed with last update

File: .github/settings.yml (hunk 1 above)
Comment (blackfalcon) [reply .github/settings.yml]: Addressed

File: docs/api.md [hunk 9]
```diff
@@ -10,4 +10,11 @@ auro-avatar provides users a way to ...
 | `ariaVisible` | `ariaVisible` | `Boolean` | null    | true value reveals content to screen reader      |
 | `code`        | `code`        | `String`  |         | provide airport code for requested airport image |
 | `img`         | `img`         | `String`  |         | provide location or URL for image to be used     |
-| `tail`        | `tail`        | `string`  |         |                                                  |
+| `tail`        | `tail`        | `String`  |         | provide tail logo for requested airline          |
+| `type`        | `type`        | `string`  |         |                                                  |
```
Comment (jason-capsule42) [inline docs/api.md:14]: Missing comments to auto-populate this

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]
Description in API is missing.

Noting that this migration was done without the previous work for the LG format merged in first. Assuming that the PR for the LG format will be updated after this is merged in.

**Review by jason-capsule42 — APPROVED** [review]

**Pull request merged**